__tests__/app.test.js

```diff
@@ -110,58 +110,140 @@ describe("app", () => {
-        return request(app)
-          .get("/api/articles/1000/comments")
-          .expect(404)
-          .then(({ body }) => {
-            const { msg } = body;
-            expect(msg).toBe("not found");
-          });
+      describe("POST /api/articles/:article_id/comments", () => {
+        test("201: responds with newly added comment", () => {
+          return request(app)
+            .post("/api/articles/3/comments")
+            .send({
+              username: "butter_bridge",
+              body: "This is a test comment.",
+            })
+            .expect(201)
+            .then(({ body }) => {
+              expect(body).toHaveProperty("comment_id", 19);
+              expect(body).toHaveProperty("article_id", 3);
+              expect(body).toHaveProperty("author", "butter_bridge");
+              expect(body).toHaveProperty("votes", 0);
+              expect(body).toHaveProperty("created_at", expect.any(String));
+              expect(body).toHaveProperty("body", "This is a test comment.");
+            });
+        });
+        test("404: responds with not found if the article does not exist", () => {
+          return request(app)
+            .post("/api/articles/3000/comments")
+            .send({
+              username: "butter_bridge",
+              body: "This is a test comment.",
+            })
+            .expect(404)
+            .then(({ body }) => {
+              const { msg } = body;
+              expect(msg).toBe("not found");
+            });
+        });
+        test("400: responds with bad request when passed invalid article_id", () => {
+          return request(app)
+            .post("/api/articles/hello/comments")
+            .send({
+              username: "butter_bridge",
+              body: "This is a test comment.",
+            })
+            .expect(400)
+            .then(({ body }) => {
+              const { msg } = body;
+              expect(msg).toBe("bad request");
+            });
+        });
+        test("400: responds with bad request when passed incorrect request body", () => {
+          return request(app)
+            .post("/api/articles/3/comments")
+            .send({
+              hello: "butter_bridge",
+              goodbye: "This is a test comment.",
+            })
+            .expect(400)
+            .then(({ body }) => {
+              const { msg } = body;
+              expect(msg).toBe("bad request");
+            });
+        });
+        test("400: responds with bad request when passed incorrect value types in request body", () => {
+          return request(app)
+            .post("/api/articles/3/comments")
+            .send({
+              username: 3,
+              body: {key: "value"},
+            })
+            .expect(400)
+            .then(({ body }) => {
+              const { msg } = body;
+              expect(msg).toBe("bad request");
+            });
+        });
```

Comment:
Great tests so far. Would also be good to see a test for a status 201, where your endpoint ignores unnecessary properties